### PR TITLE
Gateway API: update to v0.4.3, add validating webhook

### PR DESCRIPTION
**Pull request by skriss**
Updates the version of Gateway API to
v0.4.3 and adds the validating webhook
from that version to Contour's example
YAML files and E2E tests.

Closes https://github.com/projectcontour/contour/issues/3807.

Signed-off-by: Steve Kriss <krisss@vmware.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -230,10 +232,17 @@ generate-crd-yaml:
 	@echo "Generating Contour CRD YAML documents..."
 	@./hack/generate-crd-yaml.sh
 
-.PHONY: generate-gateway-crd-yaml
-generate-gateway-crd-yaml:
+# TODO switch to version-specific YAML for the validating webhook
+# once v0.5.0 comes out (v0.4.x is using a certgen tool that's 
+# incompatible with Kubernetes 1.22+).
```
Comment (skriss) [inline Makefile:237]: I'll file an issue for this; we may get a v0.4.3 with the fix too before our next Contour release.

File: test/e2e/gateway/invalid_forward_to_test.go [hunk 2]
```diff
@@ -63,28 +63,6 @@ func testInvalidForwardTo(namespace string) {
 							},
 						},
 					},
-
```
Comment (skriss) [inline test/e2e/gateway/invalid_forward_to_test.go:66]: This route rule without a port in the backend ref is now rejected by the webhook, so dropping it to allow the test to pass.

**Review by sunjayBhatia — APPROVED** [review]

**Comment by skriss** [comment]
Thinking we should hold this one until there is a release-tagged image for the webhook, otherwise we're using a `latest` image tag which will not be stable over time (https://github.com/projectcontour/contour/blob/5255e0f2f8d51f06abc7a49e3e71aa95e86467ef/examples/gateway/01-admission_webhook.yaml#L64).

**Comment by youngnick** [comment]
When Rob cuts a 0.4.3 Gateway API release, we will get a versioned webhook, so I agree we should hold until then.

File: Makefile (hunk 1 above)
Comment (skriss) [reply Makefile]: Now using versioned webhook YAML.

**Comment by skriss** [comment]
> When Rob cuts a 0.4.3 Gateway API release, we will get a versioned webhook, so I agree we should hold until then.

This is now updated for Gateway API v0.4.3

**Pull request merged**